tasks/sep/fp-recovery/005-set-game-implementation/justfile

```diff
@@ -0,0 +1,91 @@
+set positional-arguments
+
+# default recipe to display help information
+default:
+  @just --list
+
+clean:
+  rm -rf out
+
+prep l1='' l2='':
+  #!/usr/bin/env bash
+  set -euo pipefail
+  mkdir out
+  sed -e 's/<L2_CHAIN>/{{l2}}/g' -e 's/<L1_CHAIN>/{{l1}}/g' templates/README.md > out/README.md
+  cp templates/input.json out/input.json
+  cp templates/VALIDATION.md out/VALIDATION.md
+  cp templates/NestedSignFromJson.s.sol out/NestedSignFromJson.s.sol
+
+  if [[ "{{l1}}" == "sepolia" ]]; then
+    cat > out/.env << EOF
+  ETH_RPC_URL="https://ethereum-sepolia.publicnode.com"
+  COUNCIL_SAFE=0xf64bc17485f0B4Ea5F06A96514182FC4cB561977
+  FOUNDATION_SAFE=0xDEe57160aAfCF04c34C887B5962D0a69676d3C8B
+  EOF
+  elif [[ "{{l1}}" == "mainnet" ]]; then
+    cat > out/.env << EOF
+  ETH_RPC_URL="https://ethereum.publicnode.com"
+  COUNCIL_SAFE=0xc2819DC788505Aac350142A7A707BF9D03E3Bd03
+  FOUNDATION_SAFE=0x847B5c174615B1B7fDF770882256e2D3E95b9D92
+  EOF
+  else
+    echo "Unknown L1 chain."
+    exit 1
+  fi
+
+  CONFIG_URL="https://raw.githubusercontent.com/ethereum-optimism/superchain-registry/refs/heads/main/superchain/configs/{{l1}}/{{l2}}.toml"
+  CONFIG_TOML=$(curl -s --show-error --fail "${CONFIG_URL}")
+  SYSTEM_CONFIG=$(echo "${CONFIG_TOML}" | yq -p toml .addresses.SystemConfigProxy)
+  OWNER_SAFE=$(echo "${CONFIG_TOML}" | yq -p toml .addresses.ProxyAdminOwner)
+
+  cat >> out/.env << EOF
+  SAFE_NONCE=""
+  L1_CHAIN_NAME="{{l1}}"
+  L2_CHAIN_NAME="{{l2}}"
+  SYSTEM_CONFIG="${SYSTEM_CONFIG}"
+  OWNER_SAFE="${OWNER_SAFE}"
+  EOF
+
+# Generate the `input.json` with the game type and implementation to set.
+set-implementation gameType='' implAddr='':
```

Comment:
Should we do some basic validation on `implAddr` to verify that it is a fault dispute game, e.g. make a single call that should either not revert, or return some known value

Reply:
The `NestedSignFromJson` checks call quite a range of methods on this contract. I've generally deferred most validation to there rather than trying to do it in bash, especially given that regenerating the task from scratch is so easy.